components/net/http_loader.rs

```diff
@@ -537,6 +537,23 @@ pub fn http_fetch(request: Rc<Request>,
                   done_chan: &mut DoneChannel,
                   context: &FetchContext)
                   -> Response {
+    // Bad port blocking - https://fetch.spec.whatwg.org/#block-bad-port
+    match request.url().port() {
+        Some(port) => {
+            let is_ftp: bool = request.url().scheme() == "ftp" && (port == 20 || port == 21);
```

Comment:
Just added this because it made things a bit clearer for me, but it's definitely not needed.